core/iwasm/compilation/aot.h

```diff
@@ -177,6 +180,19 @@ typedef struct AOTImportGlobal {
     uint32 size;
     /* The data offset of current global in global data */
     uint32 data_offset;
+#if WASM_ENABLE_WAMR_COMPILER != 0 || WASM_ENABLE_JIT != 0
+    /*
+     * The data size and data offset of a wasm global may vary
+     * in 32-bit target and 64-bit target, e.g., the size of a
+     * GC obj is 4 bytes in the former and 8 bytes in the
+     * latter, the AOT compiler needs to use the correct data
+     * offset according to the target info.
+     */
+    uint32 size_64bit;
+    uint32 size_32bit;
+    uint32 data_offset_64bit;
```

Comment:
Can we merge them? We known the target info at compile time

Reply:
Seems some functions doesn't have access to target info stored in comp_ctx, if we merge these field, seems we need to modify many API signatures

Reply:
There may be two possible solutions: one is passing `option.target_arch` to `aot_create_comp_data`, and parsing the target's pointer size according to `option.target_arch`, for example, check whether it contains "x86_64"/"aarch64"/"riscv64", but I am not sure whether it is enough; the other is to set global data size/offset to 0 in `aot_create_comp_data`, and update them again in `aot_create_comp_context` after target info is resolved.

Reply:
OK, I think we can keep this solution first and refine the usage of target_arch.